droid-results/src/main/java/uk/gov/nationalarchives/droid/profile/CsvItemWriter.java

```diff
@@ -246,7 +251,53 @@ private static String nullSafeDate(Date date, FastDateFormat format) {
     private static String toFileName(String name) {
         return FilenameUtils.getName(name);
     }
-    
+
+    private String toFilePath(URI uri) {
+        if (uri == null) {
+            log.warn("[URI not set]");
+            return "";
+        }
+        if (FILE_URI_SCHEME.equals(uri.getScheme())) {
+            return Paths.get(uri).toAbsolutePath().toString();
+        }
+
+        // for URIs that have other than "file" scheme
+        String result = java.net.URLDecoder.decode(uri.toString()).replaceAll("file://", "");
+        result = result.replace("/", File.separator);
+
+        // Handle substitution of 7z
+        final String sevenZedIdentifier = "sevenz:";
+        if (result.startsWith(sevenZedIdentifier)) {
+            result = "7z:" + result.substring(sevenZedIdentifier.length());
+        }
+
+        return result;
+    }
+
+/*
+        if (uri == null)
+            return ;
+        if ("file".equals(uri.getScheme()))
+            return Paths.get(uri).toAbsolutePath().toString();
+
+        String result = java.net.URLDecoder.decode(uri.toString()).replaceAll("file://", "");
+
+        // Handle substitution of 7z
+        final String sevenZedIdentifier = "sevenz:";
+        if (result.startsWith(sevenZedIdentifier))
+            result = "7z:" + result.substring(sevenZedIdentifier.length());
+
+        // Handle substitution of gz
+        final String gzIdentifier = "gz:";
+        if (result.startsWith(gzIdentifier))
+            result = "gzip:" + result.substring(gzIdentifier.length());
+
+        return result;
+
+ */
```

Comment:
did you intend to have this commented section still in the code?

Reply:
oops, well spotted. Yes that needs to go away.

Reply:
done and pushed the change